lib/spack/spack/cmd/external.py

```diff
@@ -168,6 +174,18 @@ def packages_to_search_for(
         select_re = re.compile("|".join(parts))
         result = [x for x in result if not select_re.search(x)]
 
+    if dependencies_only:
+        deps_to_find = set()
+        # if we're writing dependencies "result" is now our source of
+        # packages to dervice dependencies from
+        for pkg in result:
+            pkg_cls = spack.repo.PATH.get_pkg_class(pkg)
+            deps_to_find = deps_to_find | set(pkg_cls.possible_dependencies(
+                    False, False
```

Comment:
Can you use kwarg assignment? I think this is setting `transitive=False, expand_virtuals=False`, which I think is reasonable (because if these are true then this function can take quite a while)

Reply:
Yes, it's setting both to false for performance reasons and because, as we're grabbing the immediate child nodes from the system (or at least trying to) of the root node, we just don't need any of the transitive dependencies.